sources/engine/Stride.Graphics/OpenGL/GraphicsDevice.OpenGL.cs

```diff
@@ -658,23 +658,23 @@ protected unsafe void InitializePlatformDevice(GraphicsProfile[] graphicsProfile
 #endif
 
 #if STRIDE_UI_SDL
-            gameWindow = (Stride.Graphics.SDL.Window)windowHandle.NativeWindow;
+            gameWindow = windowHandle?.NativeWindow as SDL.Window ?? new SDL.Window("");
```

Comment:
What's this new window for ?

Reply:
Dump window is created to initialize the SDLContext which is then used for the OpenGL api as a context (see 699 line).
The author assumed when casting that the `windowhandle` object was not `null` but it wasn't, so I handled it

Reply:
For opengl on windows this would load in sdl when using a winforms window then ?

Also, shouldn't it be initialized/set in a more appropriate spot then ? I.E.: ensure window is init first and has a native window handle, then this scope is called and retrieves the window without creating one ?

Reply:
When it comes to Windows, from what I've noticed, Stride Asset Compiler uses DirectX on Windows to compile assets and I haven't changed that. My change only applies to Linux (or MacOs if someone has decided to support it)

Reply:
As for the second question, this window object is used **before** the game starts so it will always be null. Also, I am not sure it even creates "real" window because Silk has command for this : SDL.CreateWindow()

Reply:
If you observe the flow of the program you will come to the conclusion that this is the earliest place where you can handle the `null` value is exactly in `InitializePlatformDevice` because only the `GraphicsDevice.Recreate` method which is common code for all platforms

Reply:
Checked Q#1 and games running on windows can enter this scope, but setting OpenGL as the backend forces SDL, this should be fine.
As for Q#2, just to clarify, the new window workaround here is specifically for the asset compiler, a game under any platform would have a valid native window value there ?

Reply:
Correct! Handling null is only for the purpose of AssetCompiler, more specifically `GraphicsDevice.New()` command - for DX11 it passes windowHandle with null and does nothing, for OpenGL it passes windowHandle and may throw NullRefException. 
When It comes to Stride runtime It shouldnt affect anything because creating game goes by `GraphicsDeviceManager.CreateDevice(...WindowHandle window)`

Reply:
Apparently no one predicted  it would be possible to run on Linux in the future 😅

Reply:
Anyways, If you have any suggestions feel free to give any ideas

Reply:
Not sure why the OGL stuff requires a window when the D3d one doesn't, but that's best left to a proper refactor I guess.
Can you split the ternary into a specific line with a comment mentioning that this is a workaround specifically for the asset compiler ?

Reply:
@Eideren Sure, Done!

Reply:
Correct me if I'm wrong, but I think OpenGL has always needed a window to establish its context (the surface where to draw). Direct3D also needed it up until D3D9, but if I remember it correctly, it was D3D10 or 11 that started to allow setting up a swapchain without an associated HWND.
As Xenko supported both D3D9 and OpenGL, that may be the reason for requiring a window. Now that D3D9 is deprecated and D3D11 no longer needs it, only OpenGL is left with that requirement (as Vulkan can also render without a window).